src/cdi/CDI.cc

```diff
@@ -267,8 +267,10 @@ void CDI::integrate_Planckian_Spectrum(std::vector<double> const &bounds,
   planck.resize(groups, 0.0);
 
   // nu/T must be < numeric_limits<double>::max().  So, if T < nu*min(), then
-  // return early with planck == 0.0. This avoids a possible divide by zero.
-  if (T <= bounds[0] * std::numeric_limits<double>::min())
+  // return early with planck == 0.0. This avoids a possible divide by zero but
+  // it doesn't catch the case where bounds[0] == 0.0 and T is a denormal so
+  // make it the more restrictive T <= numeric_limits<double>::min()
+  if (T <= std::numeric_limits<double>::min())
```

Comment:
I think this change breaks the previous logic.  I believe the issue was the case where `bounds[0]` is large so that `bounds[0]/T` would overflow when T>min.  In this scenario, your change makes the check less restrictive and enables the possibility of overflow.

However, your change brings up the case where bounds[0] is small and `bounds[0]*min` underflows.  Maybe we need are more complex set of logic.  Something like:

```c++
if(bounds[0]>1.0) { 
  if (T <= bounds[0] * std::numeric_limits<double>::min())
    return;
} else {
  if (T <= std::numeric_limits<double>::min()) // can this ever occur?
    return;
}
```